### ISSUE TITLE
Allow programmatically submit

### ISSUE THREAD

**Issue by gustavoguichard** [SeasonedSoftware/remix-forms#17]

**Comment by azzzy** [comment]
I also need this in order to be able to reset the form.
Maybe a low hanging fruit will be to simply add a an optional `Ref<HtmlFormElement>` prop to the component? This won't lead to a breaking change, unlike using `forwardRef`. This is how I am monkey-patching at the moment.
[form-ref.txt](https://github.com/SeasonedSoftware/remix-forms/files/9639374/form-ref.txt)

**Comment by danielweinmann** [comment]
@azzzy, have you tried using the `onTransition` prop for resetting the form? We do that [in this example](https://remix-forms.seasoned.cc/examples/forms/use-fetcher).

Since we added `onTransition` I haven't had any other use case where accessing the form imperatively is necessary. How about your use cases?

**Comment by azzzy** [comment]
@danielweinmann, this works nicely, somehow I missed it in the documentation. Thanks for the hint!

**Comment by diogob** [comment]
@danielweinmann talking to @gustavoguichard we found a use case for this, when one wants to imperatively trigger the form submission (let's say when a verification code is typed). But instead of relaying the form ref we could expose a submit function for the form children function. Thoughts?

**Comment by paulm17** [comment]
I needed this ability myself.  I have a complex component where it's a dialog modal and it has it's own onClick methods and where the form resides in it's body.  There is no way to connect the forms handleSubmit and modal onClick.  

I have a really old version of remix-forms which I forked a while ago and not had the time to update it.  Well, because it works so well.  😄 

All I ended up doing was the following to the Form component for it to support a ref property:

```
function Form<Schema extends SomeZodObject>(
  {
    mode = "onSubmit",
    ...
    ...props
  }: FormProps<Schema>,
  ref,
) {
   return (
    <form
      ref={ref}
      ...
    />
}

export default forwardRef(Form)
```

Then the following to execute programmatically:

```
if (formRef.current) {
   ;(formRef.current as EventTarget).dispatchEvent(
      new Event("submit", { cancelable: true, bubbles: true }),
      )
}
```

**Comment by diogob** [comment]
@paulm17 I'm curious to know if the solution in #136 would solve your use case. We have exposed a submit function that triggers the submit event, but that is only being exposed in the form children function.

**Comment by paulm17** [comment]
@diogob Although I took a look at the solution in #136 I am not quite sure.  That said, once I have finished the current work that I am going through.  I will be bringing up my fork of remix-forms to the latest and then I'll see where things stand.  Maybe in a week or so, I'll then be able to comment here on any improvements or it's perfect for my use case.  😄.  

Many thanks for being reactive and making a PR so quickly for this amazing library.  Looking forward to comes next in '23.  🚀
